### ISSUE TITLE
Compatibility with a local Git server?

### ISSUE THREAD

**Issue by nununo** [larshp/abapGit#306]
Hello,

Since my company's code cannot be posted to the outside world due to security policies I cannot use GitHub or any other online service to host my repositories.

So I was trying to find a network server where I could install Git Server making it available in the company's local network.

But then I read in the README that abapGit still only supports GitHub, GitLab, Assembla and Bitbucket.

Does it really not work with a local Git Server? If not, will it ever? If so, when? :)

Thanks in advance,
Nuno

**Comment by larshp** [comment]
I've tested abapGit with those services, however I think it works with many other git servers, as abapGit communicates with the Git server via the raw git protocol. The Git server just have to support communication via HTTPS/SSL.

Feel free to raise an issue if you find a git server that doesnt work with abapGit.

Also see https://github.com/larshp/abapGitServer which is a git server implemented in ABAP. Its functionality is quite restricted, and most likely contains a lot of bugs :) But it is something that might be easier to install since you already have a SAP server on your network.
I've tested abapGitServer with abapGit, however if you expect 1000s of objects and 1000s of commits then I'd guess that abapGitServer is too slow in its current implementation.

**Comment by larshp** [comment]
@nununo can we close this issue?

**Comment by nununo** [comment]
Done.

**Comment by nununo** [comment]
I finally managed to run abapGitServer.

But I bumped into something which I may not be  able to solve. The local machine running abapGitServer doesn't seem to have a valid SSL certificate as the https prefix in the URL is crossed out. It makes sense since it is just a DEV system. I don't see how I could install an SSL certificate since I have no access to it. 

Does that mean I'm stuck? :(

**Comment by larshp** [comment]
If running on local machine you can try changing the "Logon Data" in transaction SICF under service ZGIT. Perhaps "Security Requirement" is set to SSL

If it is more of a shared development system, I'd recommend having your SAP Basis people to look into fixing the SSL certificate.

**Comment by nununo** [comment]
Hum...

SICF seems to be ok:
![image](https://cloud.githubusercontent.com/assets/532049/18248702/c9bf4e56-7371-11e6-9f2c-67e0cdd18a2f.png)

And I can access the service's gui and I already successfully created my first repository. The problem is when I try to clone it with abapGit.

The .git URL provided by abapGitServer starts with HTTP://
But then when I try to add it I still get the error.

I assumed this was because you said above that SSL is a requirement (something I only realized after the downgrade). Is it a fact that SSL is required in the git server for abapGit to be able to clone it?

I'll try to investigate on how I could ask for the SSL certificate to be fixed. Since it's a DEV system maybe they could install one of those free certificates from https://letsencrypt.org/
